lib/queue.rb

```diff
@@ -1,31 +1,65 @@
+require_relative 'linked_list'
+
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = @back = -1
+    @counter = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    # if @front == @back
+    #   raise NoSpaceError
+    # end
+
+    @counter += 1
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    #Decrease front
+    if @front == -1 
+      raise QuequeEmptyError
+    end
+
+    @counter -= 1
+    data = @store[@front] 
+    @store[@front] = nil
+
+    if @front == @back
+      @front = -1
+      @back = -1
+    elsif @front == (@store.length - 1)
+      @front = 0
+    else
+      @front = @front + 1
+    end
+
+    return data
   end
 
   def front
     raise NotImplementedError, "Not yet implemented"
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return @counter
   end
 
   def empty?
-    raise NotImplementedError, "Not yet implemented"
+    return @counter == 0
   end
 
   def to_s
-    return @store.to_s
+    string = "["
+    string += @store.filter {|i| i != nil}.join(", ")
+    return string + "]"
   end
```

Comment:
This works, but it's quite expensive.  It also doesn't preserve the order of the buffer.  If the back has wrapped around the buffer, elements in the rear of the buffer will show up before the 1st part of the buffer.